### ISSUE TITLE
Extending Support for Syntax Highlighting via Codehilite 

### ISSUE THREAD

**Issue by ss4328** [erwinmatijsen/django-markdownify#32]
Extending Issue #6 

### Background
Looks like fenced_code extension now works but looks like the Syntax highlighting is still not working. 

I've done the following in my Settings.py file for the project 

https://github.com/ss4328/django_dev_blog

```python

MARKDOWNIFY = {
    "default": {
        "MARKDOWN_EXTENSIONS": [
            'markdown.extensions.fenced_code',
            'markdown.extensions.extra'
        ],
        "WHITELIST_STYLES": [
            'color',
            'font-weight',
        ]
    }
}

MARKDOWNIFY_WHITELIST_TAGS = [
'a',
'abbr',
'acronym',
'b',
'blockquote',
'em',
'i',
'li',
'ol',
'p',
'strong',
'ul',
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'img',
'code'
]

MARKDOWNIFY_WHITELIST_ATTRS = [
'href',
'src',
'alt',
]

MARKDOWNIFY_WHITELIST_STYLES = [
'color',
'font-weight',
]

MARKDOWNIFY_WHITELIST_PROTOCOLS = [
'http',
'https',
]

```

### Problem
Either we're not supporting syntax-highlighting out of the box, or looks like I'm missing something rudimentary. The code block is being recognized (as evident by the pink color), but looks like syntax highlighting is not working (as evident by missing new lines, and missing background styling)

![Screen Shot 2021-03-23 at 9 09 22 PM](https://user-images.githubusercontent.com/25440531/112238992-1b4df980-8c1c-11eb-82de-59542db0f972.png)

![Screen Shot 2021-03-23 at 9 09 32 PM](https://user-images.githubusercontent.com/25440531/112239010-22750780-8c1c-11eb-97e0-2374f800db21.png)

I've tried to follow 
- https://python-markdown.github.io/extensions/fenced_code_blocks/
and 
- https://python-markdown.github.io/extensions/code_hilite/#setup

I also Installed pygments and generated+loaded CSS but syntax highlighting is missing.

**Comment by erwinmatijsen** [comment]
I see some issues with your setup:

1. You are mixing old-style settings (pre-0.90 release) and new-style settings.
2. You are not including the `markdown.extensions.codehilite` in the `MARKDOWN_EXTENSIONS` list.
3. Pygment is adding the styles via `classes`, `spans` and `divs`, so you need to whitelist those `tags` and `attrs` in `WHITELIST_TAGS` and `WHITELIST_ATTRS`.

So for a complete settings-example (only the relevant parts):

```
MARKDOWNIFY = {
    "default": {
        "MARKDOWN_EXTENSIONS": [
            'markdown.extensions.extra',  # This includes fenced code, and others
            'markdown.extensions.codehilite',
        ],
        "WHITELIST_TAGS": [
            'code',
            'pre',
            'span',
            'div',
        ],
        "WHITELIST_ATTRS": [
            'class',
            ],
    }
}
```

Now the code should be highlighted.

**Comment by ss4328** [comment]
Hey, @erwinmatijsen, Thanks for the input with settings. Actually, it turns out that I didn't change the Bleach setting to false (defaults to True). As soon as I fixed it, the code highlight worked like a charm!

Leaving this working code block for settings.py here in case anyone else stumbles on the same issue.

```python

MARKDOWNIFY = {
    "default": {
        "MARKDOWN_EXTENSIONS": [
            'markdown.extensions.fenced_code',
            'markdown.extensions.extra',
            'markdown.extensions.codehilite',
        ],
        "WHITELIST_STYLES": [
            'color',
            'font-weight',
        ],
        "WHITELIST_TAGS": [
            'code',
            'pre',
            'span',
            'div',
            'a',
            'abbr',
            'acronym',
            'b',
            'blockquote',
            'em',
            'i',
            'li',
            'ol',
            'p',
            'strong',
            'ul',
            'h1',
            'h2',
            'h3',
            'h4',
            'h5',
            'img',
        ],
        "WHITELIST_ATTRS": [
            'class',
            'href',
            'src',
            'alt',
            ],
        "WHITELIST_PROTOCOL": [
            'http',
            'https',
        ],
        "BLEACH": False
    }
}
```

Thanks for the help.

**Comment by erwinmatijsen** [comment]
Setting `BLEACH` to `False` has the same effect as allowing all `tags` and `attrs`, that is why it now is working. If you want a more restricted approach, you can leave `BLEACH` set to `True` and allow the right `tags` and `attrs`. Your last example should work. 

Good luck!

**Comment by zagorsky** [comment]
I also had an issue where syntax highlighting wasn't working in a code block, but it turned out my issue was that I needed to use Pygments to generate a CSS file with all the CodeHilite classes, and link to that file in my HTML template header, following [these instructions for the CodeHilite Python-Markdown extension](https://python-markdown.github.io/extensions/code_hilite/).

Do I have that right?  I'm posting this to either find out that I did an extraneous step, or to help the next person who reads this issue looking for a solution.
